### PR TITLE
mail-filter/rspamd: Version bump to 1.8.0

### PR DESCRIPTION
**Pull request by croessner**

### CODE REVIEW

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+if [[ ${PV} == *9999 ]] ; then
+	SCM="git-r3"
+	EGIT_REPO_URI="https://github.com/vstakhov/rspamd.git"
+fi
+
+inherit cmake-utils user systemd pax-utils ${SCM}
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: Please sort whenever possible.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	SRC_URI="https://github.com/vstakhov/rspamd/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE="cpu_flags_x86_ssse3 fann gd jemalloc +jit libressl pcre2 +torch"
+REQUIRED_USE="torch? ( jit )"
+
+RDEPEND="!libressl? ( dev-libs/openssl:0=[-bindist] )
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: Also sort here, first unconditional deps by package, then by condition.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	libressl? ( dev-libs/libressl:0= )
+	fann? ( sci-mathematics/fann )
+	pcre2? ( dev-libs/libpcre2[jit=] )
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: Do you really need USE=jit to match between the packages? That looks a bit weird.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	!pcre2? ( dev-libs/libpcre[jit=] )
+	jit? ( dev-lang/luajit:2 )
+	jemalloc? ( dev-libs/jemalloc )
+	dev-libs/libevent
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: If you're linking to it, you certainly want `:=` to force rebuilds on ABI changes.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	net-libs/libnsl
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	<dev-util/ragel-7.0
+	sys-apps/file
+	gd? ( media-libs/gd[jpeg] )
+	dev-libs/icu
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: Another `:=` candidate.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	cpu_flags_x86_ssse3? ( dev-libs/hyperscan )"
+DEPEND="dev-util/ragel
+	${RDEPEND}"
+
+QA_MULTILIB_PATHS="usr/lib/rspamd/.*"
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: I don't think this is necessary anymore.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	dev-libs/libevent
+	net-libs/libnsl
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	<dev-util/ragel-7.0
+	sys-apps/file
+	gd? ( media-libs/gd[jpeg] )
+	dev-libs/icu
+	cpu_flags_x86_ssse3? ( dev-libs/hyperscan )"
+DEPEND="dev-util/ragel
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: I'm a bit confused. Does it mean you can build with ragel > 7.0 but you need to install older version to use it afterwards?

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+	newinitd "${FILESDIR}/rspamd.init-r5" rspamd
+
+	# Remove mprotect for JIT support
+	if use jit; then
+		pax-mark m "${ED}"/usr/bin/rspamd-* "${ED}"/usr/bin/rspamadm-* || die
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: I think `pax-mark` doesn't need `||die`.

File: mail-filter/rspamd/rspamd-1.8.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+	fi
+
+	keepdir /var/lib/rspamd
+	keepdir /var/log/rspamd
+
+	fowners rspamd:rspamd /var/lib/rspamd /var/log/rspamd
```
Comment (mgorny) [inline mail-filter/rspamd/rspamd-1.8.0.ebuild]: Why reown it after creating when you can use `diropts` to set permissions for `keepdir`?

**Comment by djc** [comment]
1.8.1 has been released. @croessner ping??

**Pull request closed without merging**